File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.cpp
#include <vector>
#include <iostream>

using namespace std;

int solution(vector<int> nums) {
    vector<int> prime(3000, 1);
    prime[0] = 0;
    prime[1] = 0;
    
    for (int i = 2; i * i <= 3000; i++) {
        if (prime[i] == 1) {
            for (int j = i * i; j <= 3000; j += i) {
                prime[j] = 0;
            }
        }
    }
    
    int answer = 0;
    for (int i = 0; i < nums.size() - 2; i++) {
        for (int j = i + 1; j < nums.size() - 1; j++) {
            for (int k = j + 1; k < nums.size(); k++) {
                int val = nums[i] + nums[j] + nums[k];
                
                if (prime[val]) answer++;
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.py
from itertools import combinations

def solution(nums):
    prime = [1] * 3000
    prime[0] = prime[1] = 0
    
    for i in range(2, int(3000 ** 0.5) + 1):
        if prime[i] == 1:
            for j in range(i * i, 3000, i):
                prime[j] = 0
    
    return sum(prime[sum(comb)] for comb in combinations(nums, 3))